esteid/mixins.py

```diff
@@ -52,7 +55,16 @@ def report_error(self, e: EsteidError):
     def handle_user_cancel(self):
         pass
 
-    def handle_errors(self, e: Exception, stage="start"):
+    def on_failure(self, e: Exception, stage: SessionStageType, request: HttpRequest):
+        """
+        Hook called when exception occurs during authentication/signing.
+        It runs before "handle_errors" and allows to perform logging, auditing, or other side effects.
+        """
+        pass
+
+    def handle_errors(self, e: Exception, stage: SessionStageType = "start", request: HttpRequest = None):
```

Comment:
Why do we need to have a separate `on_failure` if we can just override `handle_errors` and do 

```
def handle_errors(...):
    do_whatever_on_failure_needs_to_do()
    return super().handle_errors(...)
```